analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/MetricsAnalyzerTest.cs

```diff
@@ -87,6 +87,25 @@ public class MetricsAnalyzerTest
                         // There should be no metrics messages for the cshtml files.
                         messages.Select(x => Path.GetFileName(x.FilePath)).Should().BeEquivalentTo("_Imports.razor"));
 
+        [TestMethod]
+        public void VerifyMetrics_CSharp12() =>
+            CreateBuilder(false, CSharp12FileName)
+                .WithAdditionalFilePath(AnalysisScaffolding.CreateSonarProjectConfig(TestContext, ProjectType.Product))
+                .VerifyUtilityAnalyzer<MetricsInfo>(messages =>
+                {
+                    messages.Should().ContainSingle();
+                    var metrics = messages.Single();
+                    metrics.ClassCount.Should().Be(1); // no changes
+                    metrics.CodeLine.Should().HaveCount(12);
+                    metrics.CognitiveComplexity.Should().Be(1); // no changes
+                    metrics.Complexity.Should().Be(3); // no changes
+                    metrics.ExecutableLines.Should().HaveCount(3); // 5, 7, 9
+                    metrics.FunctionCount.Should().Be(2); // no changes
+                    metrics.NoSonarComment.Should().BeEmpty();
+                    metrics.NonBlankComment.Should().ContainSingle();
+                    metrics.StatementCount.Should().Be(3);
```

Comment:
Same as for `ExecutableLines`.

Reply:
Same as ExecutableLines, fields assigned to primary ctor params are not increasing the StatementCount